homeassistant/components/accuweather/sensor.py

```diff
@@ -413,34 +392,42 @@ def __init__(
             self._attr_unique_id = (
                 f"{coordinator.location_key}-{description.key}".lower()
             )
-        self._attr_native_unit_of_measurement = description.native_unit_of_measurement
-        if self.coordinator.hass.config.units is METRIC_SYSTEM:
-            self._unit_system = API_METRIC
-            if metric_unit := description.metric_unit:
-                self._attr_native_unit_of_measurement = metric_unit
-        else:
-            self._unit_system = API_IMPERIAL
-            if us_customary_unit := description.us_customary_unit:
-                self._attr_native_unit_of_measurement = us_customary_unit
         self._attr_device_info = coordinator.device_info
-        if forecast_day is not None:
-            self.forecast_day = forecast_day
+        self.forecast_day = forecast_day
 
     @property
     def native_value(self) -> StateType:
         """Return the state."""
-        return self.entity_description.value_fn(self._sensor_data, self._unit_system)
+        return self.entity_description.value_fn(self._sensor_data)
 
     @property
     def extra_state_attributes(self) -> dict[str, Any]:
         """Return the state attributes."""
+        if self.forecast_day is not None:
+            return self.entity_description.attr_fn(self._sensor_data)
+
         return self.entity_description.attr_fn(self.coordinator.data)
 
+    @property
+    def suggested_unit_of_measurement(self) -> str | None:
+        """Return the unit which should be used for the sensor's state.
+
+        Entity units with device_class WIND_SPEED are not automatically converted for
+        the US CUSTOMARY unit system, so we return the suggested unit for them.
+        """
+        if (
+            self.entity_description.device_class == SensorDeviceClass.WIND_SPEED
+            and self.hass.config.units == US_CUSTOMARY_SYSTEM
+        ):
+            return UnitOfSpeed.MILES_PER_HOUR
+
+        return None
+
```

Comment:
This relies on static data. You shouldn't need to check the value on every update.
Maybe move this to `async_added_to_hass`?

```suggestion
    async def async_added_to_hass(self) -> None:
        if (
            self.entity_description.device_class == SensorDeviceClass.WIND_SPEED
            and self.hass.config.units == US_CUSTOMARY_SYSTEM
        ):
            self._attr_suggested_unit_of_measurement = UnitOfSpeed.MILES_PER_HOUR
        await super().async_added_to_hass()

```

Reply:
Cancel that - it doesn't work.
It seems the `suggested_unit_of_measurement` is read before calling `async_added_to_hass`.

Reply:
We can pass suggested unit in the sensor class init by `_attr_suggested_unit_of_measurement`. I'll change that.